apps/web/src/components/settings/PushNotifications/hooks/useNotificationsRenewal.ts

```diff
@@ -0,0 +1,169 @@
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { useNotificationPreferences } from './useNotificationPreferences'
+import { useNotificationRegistrations } from './useNotificationRegistrations'
+import { useCallback, useEffect, useMemo } from 'react'
+import { useAppDispatch, useAppSelector } from '@/store'
+import { selectNotifications, showNotification } from '@/store/notificationsSlice'
+import useWallet from '@/hooks/wallets/useWallet'
+import { NotificationsTokenVersion } from '@/services/push-notifications/preferences'
+import { useIsNotificationsRenewalEnabled, useNotificationsTokenVersion } from './useNotificationsTokenVersion'
+import type { NotifiableSafes } from '../logic'
+import { flatten, isEmpty } from 'lodash'
+import useIsWrongChain from '@/hooks/useIsWrongChain'
+
+/**
+ * Hook to manage the renewal of notifications
+ * @param shouldShowRenewalNotification a boolean to determine if the renewal notification should be shown
+ * @returns an object containing the safes for renewal, the number of chains for renewal, the number of safes for renewal,
+ * the renewNotifications function and a boolean indicating if a renewal is needed
+ */
+export const useNotificationsRenewal = (shouldShowRenewalNotification = false) => {
+  const wallet = useWallet()
+  const dispatch = useAppDispatch()
+  const { safe, safeLoaded } = useSafeInfo()
+  const { registerNotifications } = useNotificationRegistrations()
+  const { getPreferences, getAllPreferences, getChainPreferences } = useNotificationPreferences()
+  const preferences = getPreferences(safe.chainId, safe.address.value)
+  const allPreferences = getAllPreferences()
+  const notifications = useAppSelector(selectNotifications)
+  const { safeTokenVersion, allTokenVersions, setTokenVersion } = useNotificationsTokenVersion()
+  const isWrongChain = useIsWrongChain()
+  const isNotificationsRenewalEnabled = useIsNotificationsRenewalEnabled()
+
+  // Check if a renewal notification is already present
+  const hasNotificationMessage = useMemo(
+    () => notifications.some((notification) => notification.groupKey === 'renewal'),
```

Comment:
`'renewal'` should be extracted as a constant and used in both places.

Reply:
Done